src/software/ai/passing/cost_function.cpp

```diff
@@ -335,27 +299,61 @@ double getStaticPositionQuality(const Field& field, const Point& position,
 }
 
 double calculateProximityRisk(const Point& point, const Team& enemy_team,
-                              double enemy_proximity_importance)
+                              const TbotsProto::PassingConfig& passing_config)
 {
-    // Calculate a risk score based on the distance of the enemy robots from the receive
+    // Calculate a risk score based on the distance of the enemy robots from the given
     // point, based on an exponential function of the distance of each robot from the
     // receiver point
-    auto enemy_robots                 = enemy_team.getAllRobots();
-    double point_enemy_proximity_risk = 1;
-    for (const Robot& enemy : enemy_team.getAllRobots())
+    if (enemy_team.getAllRobots().empty())
     {
-        double dist = (point - enemy.position()).length();
-        point_enemy_proximity_risk *= enemy_proximity_importance * std::exp(-dist * dist);
+        return 0;
     }
-    if (enemy_robots.empty())
+
+    double risk = 0;
+    for (const Robot& enemy : enemy_team.getAllRobots())
     {
-        point_enemy_proximity_risk = 0;
+        double dist_to_enemy =
+            std::max(0.0, distance(point, enemy.position()) - ROBOT_MAX_RADIUS_METERS);
+        risk += std::exp((-dist_to_enemy * dist_to_enemy) /
+                         passing_config.enemy_proximity_importance());
     }
-    return point_enemy_proximity_risk;
+    return sigmoid(risk, 1, 2);
+}
+
+double rateKeepAwayPosition(const Point& keep_away_position, const World& world,
+                            const Pass& best_pass_so_far,
+                            const Rectangle& dribbling_bounds,
+                            const TbotsProto::PassingConfig& passing_config)
```

Comment:
This function was taken directly from the keep away file. I moved it here since it seemed fitting, and now we can also visualize it like the other cost functions.

Reply:
does this get used when rating passes at all?

Reply:
No, this is just used for keep away in offense tactic. But since it's still a "cost function" and I wanted to visualize it, I decided to move it here.